CardanoSharp.Wallet.Test/HDWallet/DerivationTests.cs

```diff
@@ -0,0 +1,204 @@
+
+            // Assert
+            Assert.Equal(paymentPrv1.Key, derivation.PrivateKey.Key);
+            Assert.Equal(paymentPrv1.Chaincode, derivation.PrivateKey.Chaincode);
+
+            Assert.Equal(paymentPub1.Key, derivation.PublicKey.Key);
+            Assert.Equal(paymentPub1.Chaincode, derivation.PublicKey.Chaincode);
+        }
+
+
+        [Theory]
+        [InlineData(__mnemonic, "m/1852'")]
+        [InlineData(__mnemonic, "m/1852'/1815'")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'/0")]
+        [InlineData(__mnemonic, "m/1852'/1815'/0'/0/0")]
+        public void PartialPathDerivationTest(string words, string path)
+        {
+            var depth = path.Split("/").Slice(1).Count(); // strip master node to calc zero based depth
+
+            // create two payment addresses from same root key
+            //arrange
+            var mnemonic = new KeyService().Restore(words);
+            var rootKey = mnemonic.GetRootKey();
+            var testKey = getTestRootKey(mnemonic);
+
+            (var prv, var pub) = getKeyPairFromPath(path, rootKey);
+
+            // Act
+            // Fluent derivation API
+            var master = testKey.Derive();
+
+            var purpose = master.Derive(PurposeType.Shelley);
+            var coin = purpose.Derive(CoinType.Ada);
+            var account = coin.Derive(0);
+            var role = account.Derive(RoleType.ExternalChain);
+            var index = role.Derive(0);
+
+            if (depth == 1) AssertDerivedKeys(prv, pub, purpose);
+            if (depth == 2) AssertDerivedKeys(prv, pub, coin);
+            if (depth == 3) AssertDerivedKeys(prv, pub, account);
+            if (depth == 4) AssertDerivedKeys(prv, pub, role);
+            if (depth == 5) AssertDerivedKeys(prv, pub, index);
+        }
+
+        [Fact]
+        public void ImplicitPathDerivationTest()
+        {
+            var path = "m/1852'/1815'/0'/0/0";
+
+            // Arrange
+            var mnemonic = new KeyService().Restore(__mnemonic);
+            var rootKey = mnemonic.GetRootKey();
+            var testKey = getTestRootKey(mnemonic);
+
+            (var paymentPrv1, var paymentPub1) = getKeyPairFromPath(path, rootKey);
+
+            // Act
+            // Fluent derivation API
+            var derivation = testKey.Derive(RoleType.ExternalChain)      // implicit IRoleNodeDerivation
+                                    .Derive(0);                          // IIndexNodeDerivation
+            var derivation2 = testKey.Derive(RoleType.ExternalChain, 0); // implicit IIndexNodeDerivation
+            var derivation3 = testKey.DeriveAccount(0)                   // implicit IAccountNodeDerivation
+                                    .Derive(RoleType.ExternalChain)      // implicit IRoleNodeDerivation
+                                    .Derive(0);                          // IIndexNodeDerivation
```

Comment:
you get the drift... does this answer your question from before @nothingalike  or did I misunderstand st?

Reply:
Or are you talking about AccountDiscovery? 
Maybe I'm confused... its been a long day... 
[AccountDiscovery](https://github.com/CardanoSharp/cardanosharp-wallet/pull/17/files#diff-35b04d8d73591c795781c60ed5c8906a1856816ff7906d307a7602a63f10190aR27) could be something for `IWalletService`

Reply:
hopefully my comment above painted the scenario well enough. sometimes I'm not very good at explaining things >.<